lib/chef/provider/mount/linux.rb

```diff
@@ -0,0 +1,60 @@
+#
+# Author:: Antima Gupta (<agupta@chef.io>)
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../mount"
+
+class Chef
+  class Provider
+    class Mount
+      class Linux < Chef::Provider::Mount::Mount
+
+        provides :mount, os: "linux"
+
+        # Check to see if the volume is mounted.
+        # "findmnt" outputs the mount points with volume.
+        # Convert the mount_point of the resource to a real path in case it
+        # contains symlinks in its parents dirs.
+
+        def mounted?
+          mounted = false
```

Comment:
Instead of setting a mounted value you can just return true/false wherever you set mounted equal to true/false

Reply:
I was curious about that too, this is following the pattern of the method in the superclass that sets the value of the property at the end: https://github.com/chef/chef/blob/d0764e7849a69f5ce9b0a7bf24a7c0263a9f2c8e/lib/chef/provider/mount/mount.rb#L82

Reply:
We should nuke that pattern in the main provider as well.

Reply:
@tas50 I am setting a mounted value because we need to iterate all lines in the mount point list then only we can identify the device is mounted or not.